src/hooks/__tests__/create.test.js

```diff
@@ -0,0 +1,79 @@
+describe('create hook test', () => {
+  beforeEach(() => {
+    jest.mock('/engine/create-auth-hook', () => ({
+      createAuthHook: jest.fn(),
+      onAuthorized: jest.fn()
+    }));
+  });
+
+  afterEach(() => {
+    jest.resetModules();
+  });
+
+  test('should call create hook proper options', async () => {
+    const { create } = require('../create');
+    const { createAuthHook } = require('../../engine/create-auth-hook');
+    const options = {
+      authenticate: jest.fn(),
+      ruleServiceName: 'authorization-rules',
+      getPrincipalId: jest.fn(),
+      getArguments: jest.fn(),
+      isAuthorized: jest.fn(),
+      onAuthorized: await jest.fn(),
+      attributeLookup: {
+        principalId: jest.fn()
+      }
+    };
+
+    create(options);
+
+    expect(createAuthHook).toHaveBeenCalledWith({
+      authenticate: expect.any(Function),
+      ruleServiceName: 'authorization-rules',
+      getPrincipalId: expect.any(Function),
+      getArguments: expect.any(Function),
+      isAuthorized: expect.any(Function),
+      onAuthorized: expect.any(Function),
+      attributeLookup: {
+        principalId: expect.any(Function)
+      }
+    });
+  });
+
+  test('create onAuthorized should return true', async () => {
+    const { create } = require('../create');
+    const { createAuthHook } = require('../../engine/create-auth-hook');
+    const options = {
```

Comment:
We only need to pass in `onAuthorized` mock for this test. We could disregard all the other options and simplify this a bit.